src/com/ms/silverking/process/ProcessWaiter.java

```diff
@@ -4,39 +4,39 @@
 import java.util.concurrent.TimeUnit;
 
 public class ProcessWaiter implements Runnable {
-    private final Process   p;
-    private final SynchronousQueue<Integer>  sq;
-    
-    public static final int TIMEOUT = Integer.MAX_VALUE;
-    
-    public ProcessWaiter(Process p) {
-        this.p = p;
-        sq = new SynchronousQueue<Integer>();
-        new Thread(this, "ProcessWaiter").start();
-    }
-    
-    public int waitFor(int millis) {
-        try {
-            Integer result;
-            
-            result = sq.poll(millis, TimeUnit.MILLISECONDS);
-            if (result != null) {
-                return result;
-            } else {
-                return TIMEOUT;
-            }
-        } catch (InterruptedException ie) {
-            throw new RuntimeException("panic");
-        }
+  private final Process p;
+  private final SynchronousQueue<Integer> sq;
+
+  public static final int TIMEOUT = Integer.MAX_VALUE;
+
+  public ProcessWaiter(Process p) {
+    this.p = p;
+    sq = new SynchronousQueue<Integer>();
+    new Thread(this, "ProcessWaiter").start();
+  }
+
+  public int waitFor(int millis) {
+    try {
+      Integer result;
+
+      result = sq.poll(millis, TimeUnit.MILLISECONDS);
+      if (result != null) {
+        return result;
+      } else {
+        return TIMEOUT;
+      }
+    } catch (InterruptedException ie) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)